### PR TITLE
Add public `Config.update_from_dirs()` method

### PR DESCRIPTION
**Pull request by schlunma**
## Description

While working on making ESMValTool with the new configuration (https://github.com/ESMValGroup/ESMValCore/pull/2448) I noticed that we are missing a simple public function to update the configuration from one or more directories.

This PR adds that.

Related to #2371 

Link to documentation: https://esmvaltool--2538.org.readthedocs.build/projects/ESMValCore/en/2538/api/esmvalcore.config.html#esmvalcore.config.Config.update_from_dirs

***

## [Before you get started](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#getting-started)

- [x] [☝ Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🧪][2] The new functionality is [relevant and scientifically sound](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#scientific-relevance)
- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] Code is written according to the [code quality guidelines](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#code-quality)
- [x] [🧪][2] and [🛠][1] [Documentation](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#documentation) is available
- [x] [🛠][1] [Unit tests](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#tests) have been added
- [x] [🛠][1] Changes are [backward compatible](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#backward-compatibility)
- [x] [🛠][1] Any changed [dependencies have been added or removed](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#dependencies) correctly
- [x] [🛠][1] The [list of authors](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#list-of-authors) is up to date
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository

### CODE REVIEW

File: esmvalcore/config/_config_object.py [hunk 1]
```diff
@@ -431,6 +434,39 @@ def start_session(self, name: str) -> Session:
             session = Session(config=self.copy(), name=name)
         return session
 
+    def update_from_dirs(self, dirs: Iterable[str | Path]) -> None:
+        """Update configuration object from directories.
+
+        This will first search for all YAML files within the given directories
+        and merge them together using :func:`dask.config.collect` (values in
+        the latter directories are preferred to those in the former). Then, the
```
Comment (bouweandela) [inline esmvalcore/config/_config_object.py:442]:
```suggestion
        and merge them together using :func:`dask.config.collect` (if identical values
        are provided in multiple files, the value from the last file will be used).
        Then, the
```
Maybe rephrase like this? As far as I understand it, 'the latter' and 'the former' are typically used when there are only two options mentioned.

**Review by bouweandela — APPROVED** [review]

File: esmvalcore/config/_config_object.py [hunk 2]
```diff
@@ -431,6 +434,39 @@ def start_session(self, name: str) -> Session:
...
+            A list of directories to search for YAML configuration files.
+
+        Raises
+        ------
+        esmvalcore.exceptions.InvalidConfigParameter
+            Invalid configuration option given.
+
+        """
+        new_config_dict = self._get_config_dict_from_dirs(dirs)
+        merged_config_dict = dask.config.merge(self, new_config_dict)
```
Comment (valeriupredoi) [inline esmvalcore/config/_config_object.py:465]: when this throws a wobbly, do you want to catch and release that or filter it through our own custom exception? Just looked at both `merge` and `update` from Dask, and they're really basic, and not account for any possible exceptions, so if there are exceptions, then those will be cryptic AF

**Review by valeriupredoi — APPROVED** [review]
noicee! Just one (possible offbeat) comment from me, can merge right after Manu bothers giving me an answer to that :beer:

File: esmvalcore/config/_config_object.py (hunk 2 above)
Comment (schlunma) [reply esmvalcore/config/_config_object.py:465]: When there's an invalid option given a `InvalidConfigParameter` exception is raised in the following line (`self.update()`) via

https://github.com/ESMValGroup/ESMValCore/blob/5b283729c37879c3831ebb1428d056b9d9d7c05f/esmvalcore/config/_validated_config.py#L63-L77

Comment (valeriupredoi) [reply esmvalcore/config/_config_object.py:465]: ah my apologies Manu, too many GH invocations and tags - what can I say, am popular :grinning: Yes, am always happy with your clarifications :beer:
